### ISSUE TITLE
DataLoader に `num_workers` を設定する

### ISSUE THREAD

**Issue by kurorosu** [kurorosu/pochidetection#292]
## Summary

学習用 DataLoader の `num_workers` が未設定 (デフォルト 0) で, データ読み込みがシングルスレッドになっている.

## Background

`num_workers=0` ではメインスレッドでデータ読み込みが行われ, GPU がデータ待ちでアイドルになる. マルチワーカー化で学習スループットが大幅に向上する.

## Scope

- config に `num_workers` パラメータを追加する
- `DataLoader` に `num_workers` と `pin_memory=True` (CUDA 使用時) を設定する
- `persistent_workers=True` も検討する

## Out of Scope

- 推論パイプラインのバッチ化 (別 Issue: 6-4)

## Acceptance Criteria

- DataLoader に `num_workers` が設定される
- config から `num_workers` を制御できる
- 既存テストが通る

## Additional Context

```python
# pochidetection/scripts/common/training.py:82-93
train_loader = DataLoader(
    train_dataset,
    batch_size=batch_size,
    shuffle=True,
    collate_fn=collator,
    # num_workers 未設定
)
```

## Labels

`enhancement`

## Branch

`feat/dataloader-num-workers`
